### PR TITLE
Adjust Rmd that use h1 as top-level headings

### LINKED ISSUE #2004: Should H1 in articles be added to TOC
Example https://github.com/maelle/blabla/blob/main/vignettes/blop.Rmd
Website https://maelle.github.io/blabla/articles/blop.html

It seems to be a regression as the TOC on https://dplyr.tidyverse.org/articles/base.html looks fine

### PR DESCRIPTION
**Pull request by hadley**
Fixes #2004. Closes #2005.

### CODE REVIEW

File: R/tweak-page.R [hunk 1]
```diff
@@ -54,6 +54,12 @@ tweak_rmarkdown_html <- function(html, input_path, pkg = pkg) {
     )
   }
 
+  # If top-level headings use h1, move eveything down one level
```
Comment (maelle) [inline R/tweak-page.R:57]:
```suggestion
  # If top-level headings use h1, move everything down one level
```

File: NEWS.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # pkgdown (development version)
 
+* Vignettes/articles that use `<h1>` as top-level heading automatically have
```
Comment (maelle) [inline NEWS.md:3]: I now wonder whether it's the right decision (here and for changelogs). The page title h1 should maybe not count in the hierarchy especially as it's styled differently anyway? In Google docs there's title, subtitle and then heading 1, heading 2 etc.

File: R/tweak-page.R [hunk 3]
```diff
@@ -54,6 +54,12 @@ tweak_rmarkdown_html <- function(html, input_path, pkg = pkg) {
...
+  h1 <- xml2::xml_find_all(html, "//h1")
+  if (length(h1) > 1) {
+    tweak_section_levels(html)
```
Comment (maelle) [inline R/tweak-page.R:60]: Actually the code should not move _everything_ down one level, only the non page-header header. 
When `tweak_section_levels()` is called for news it is only called on the content of NEWS.md i.e. without any page header.

**Review by maelle — CHANGES_REQUESTED** [review]
The current version also includes the page title in the TOC. I am waiting for GitHub pages to prove my point when it updates https://maelle.github.io/blabla/articles/blip.html at last :sweat_smile:

**Comment by maelle** [comment]
See https://github.com/r-lib/pkgdown/pull/2008/files via https://stackoverflow.com/a/38266762

**Comment by maelle** [comment]
https://maelle.github.io/blabla/articles/blip.html is deployed at last and shows the page header is in the TOC.

**Comment by hadley** [comment]
I think you're saying we shouldn't adjust the page header? I agree.

**Comment by mpadge** [comment]
@hadley This PR only kinda works - the [first h1 ToC item on @maelle's vignette](https://maelle.github.io/blabla/articles/blop.html#one-cool-thing) does not list sub-(h2)-items, but instead unfolds the second h1 item. Plus it still fails for me locally, and only gives h2 items in the Toc. Likely causally related is that the `<aside>` element for the ToC is `h2`, so naturally ignores all `h1` items.

**Comment by hadley** [comment]
@mpadge that looks like it was run prior to me final changes.

**Comment by maelle** [comment]
I've updated the website and yes the first h1 in the TOC never unfolds https://maelle.github.io/blabla/articles/blop.html but I suspect it's because the section is too short. More of a bootstrap-toc/scrollspy thing @mpadge, I think. 

What was the local failure?

**Comment by mpadge** [comment]
@maelle yeah, agreed, that's likely just a scrollspy thing, so no problem here. The local failure is likely caused by `nvim-r` which doesn't always play nicely with `withr::local_options()`. I can resolve it by using `build_site(new_process = TRUE)`. Thanks both!

**Pull request merged**